apps/OboeTester/app/src/main/java/com/mobileer/oboetester/BaseAutoGlitchActivity.java

```diff
@@ -372,10 +390,80 @@ protected TestResult testInOutConfigurations() throws InterruptedException {
             testResult.setResult(result);
             mTestResults.add(testResult);
         }
-
         return testResult;
     }
 
+    protected AudioDeviceInfo getDeviceInfoById(int deviceId) {
+        AudioDeviceInfo[] devices = mAudioManager.getDevices(AudioManager.GET_DEVICES_ALL);
+        for (AudioDeviceInfo deviceInfo : devices) {
+            if (deviceInfo.getId() == deviceId) {
+                return deviceInfo;
+            }
+        }
+        return null;
+    }
+
+    protected AudioDeviceInfo getDeviceInfoByType(int deviceType, int flags) {
+        AudioDeviceInfo[] devices = mAudioManager.getDevices(flags);
+        for (AudioDeviceInfo deviceInfo : devices) {
+            if (deviceInfo.getType() == deviceType) {
+                return deviceInfo;
+            }
+        }
+        return null;
+    }
+
+    protected boolean isDeviceTypeMixed(int type) {
```

Comment:
I'm confused by the name of this function. What does mixed mean? Should USB Device return true and the others return false?

Reply:
I added comments.

Reply:
Can you push your changes?